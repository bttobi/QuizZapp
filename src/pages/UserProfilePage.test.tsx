import { Route, Routes } from 'react-router-dom';
import { UserProfilePage } from '.';
import { render, waitFor } from '../tests/test-utils';

const setup = () =>
  render(
    <Routes>
      <Route path="/profile/:userID" element={<UserProfilePage />}></Route>
    </Routes>,
    { initialEntries: ['/profile/1'] }
  );

describe('UserProfilePage component', () => {
  it('renders the component', async () => {
    const { container } = setup();
    await waitFor(() => {
      expect(container).toBeInTheDocument();
    });
  });

  it('displays fetched user stats data', async () => {
    const { getByText } = setup();
    await waitFor(() => {
      expect(getByText('All attempts count: 12')).toBeInTheDocument();
      expect(getByText('Number of created quizzes: 2')).toBeInTheDocument();
      expect(getByText('Most attempted quiz: Moj test')).toBeInTheDocument();
      expect(
        getByText('Users proficiency in best question: 100%')
      ).toBeInTheDocument();
    });
  });

  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
