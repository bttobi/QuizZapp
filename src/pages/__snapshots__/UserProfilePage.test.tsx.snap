// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserProfilePage component > matches snapshot 1`] = `
<div>
  <div
    class="w-full h-full flex flex-col justify-start items-center align-center"
    style="opacity: 0;"
  >
    <section
      class="flex flex-col gap-4 bg-primary/70 rounded-lg p-4 text-lg text-white mt-8 text-center mx-4 mb-8"
      style="transform: translateY(50rem) translateZ(0);"
    >
      <p
        class="text-3xl bg-background py-4 px-1 rounded-lg flex flex-col"
        style="text-wrap: wrap;"
      >
        <span
          class="text-2xl"
        >
          student
        </span>
        <span
          class="text-2xl"
        >
          profile stats
        </span>
      </p>
      <span>
        All attempts count: 12
      </span>
      <span>
        Number of created quizzes: 2
      </span>
      <span>
        Most attempted quiz: Moj test
      </span>
      <span>
        Highest quiz attempts: 7
      </span>
      <span>
        Overall proficiency: 83.33%
      </span>
      <span
        class="text-success"
      >
        Best quiz: Test promotora
      </span>
      <span
        class="text-success"
      >
        Best proficiency in quiz: 85.71%
      </span>
      <span
        class="text-success"
      >
        Best question: Test?
      </span>
      <span>
        Users proficiency in best question: 100%
      </span>
      <span
        class="text-danger"
      >
        Worst quiz: Testowanie
      </span>
      <span
        class="text-danger"
      >
        Worst proficiency in quiz: 81.82%
      </span>
      <span
        class="text-danger"
      >
        Worst question: Kolejne pytanie
      </span>
      <span>
        Users proficiency in worst question: 0%
      </span>
    </section>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
